robot_skills/src/robot_skills/arms.py

```diff
@@ -318,6 +322,188 @@ class GripperState(object):
+        :param arm_joints_suffix: Suffix to append to names of arm joints.
+        """
+        super(ArmJointsPiece, self).__init__(robot_name)
+
+        joint_names = self.load_param('skills/arm/joint_names')
+        self.arm_joint_names = [name + arm_joints_suffix for name in joint_names]
+        self.torso_joint_names = self.load_param('skills/torso/joint_names')
+
+    def get_arm_joint_names(self):
+        return self.arm_joint_names
+
+    def get_torso_joint_names(self):
+        return self.torso_joint_names
+
+class HandoverPiece(object):
+    """
+    Hand over an item between the arm and a human.
+    """
+    def __init__(self, arm):
+        """
+        :param arm: Arm to use for the exchange.
+        """
+        self.robot_name = arm.robot_name
+        self.side = arm.side
+
+        # Copy interface to the supplied arm.
+        arm.handover_to_human = self.handover_to_human
+        arm.handover_to_robot = self.handover_to_robot
+
+    def handover_to_human(self, timeout=10):
+        # type: (float) -> bool
+        """
+        Handover an item from the gripper to a human.
+
+        Feels if user slightly pulls or pushes (the item in) the arm. On timeout, it will return False.
+        :param timeout: timeout in seconds
+        :return: True or False
+        """
+        return self._exchange_with_human(False, timeout)
+
+    def handover_to_robot(self, timeout=10):
+        # type: (float) -> bool
+        """
+        Handover an item from a human to the robot.
+
+        Feels if user slightly pushes an item in the gripper. On timeout, it will return False.
+        :param timeout: timeout in seconds
+        :return: True or False
+        """
+        return self._exchange_with_human(True, timeout)
+
+    def _exchange_with_human(self, to_robot, timeout=10.0):
+        # type: (bool, float) -> bool
+        """
+        Exchange an item between the arm and a human.
+
+        If 'to_robot', it feels if user slightly pushes an item in the gripper.
+        If not 'to_robot', it feels if user slightly pulls or pushes (the item in) the arm.
+
+        :param to_robot: Direction of the exchange.
+        :param timeout: Timeout in seconds.
+        :return: Whether the exchange succeeded.
+        """
+        if to_robot:
+            topic = '/{}/handoverdetector_{}/toggle_human2robot'.format(self.robot_name, self.side)
+        else:
+            topic = '/{}/handoverdetector_{}/toggle_robot2human'.format(self.robot_name, self.side)
+
+        pub = rospy.Publisher(topic, std_msgs.msg.Bool, queue_size=1, latch=True)
+        pub.publish(std_msgs.msg.Bool(True))
+
+        result_topic = '/{}/handoverdetector_{}/result'.format(self.robot_name, self.side)
+        try:
+            rospy.wait_for_message(result_topic, std_msgs.msg.Bool, timeout)
+            return True
+        except rospy.ROSException as e:
+            rospy.logerr(e)
+            return False
+
+class ArmDatabasePiece(ArmPiece):
```

Comment:
Dit zijn specifiek _joint_ goals en trajectories. Zouden we die niet strakker willen koppelen aan de `ArmsJointPiece` klasse? Overigens zou daar dan de bijbehorende action client in geconstrueerd kunnen worden en kunnen daar ook de `has_joint_trajectory`, `send_joint_goals` etc. bijhoren.